onnx_caffe2/backend.py

```diff
@@ -472,20 +485,46 @@ def prepare(cls, model, device='CPU', **kwargs):
         '''
         super(Caffe2Backend, cls).prepare(model, device, **kwargs)
 
-        init_net, predict_net = cls.onnx_graph_to_caffe2_net(model.graph)
+        ws = Workspace()
         device_option = get_device_option(Device(device))
-        init_net.device_option.CopyFrom(device_option)
-        predict_net.device_option.CopyFrom(device_option)
 
+        # Directly load initializer data into blobs in workspace
+        cls._direct_initialize_parameters(
+            model.graph.initializer,
+            ws,
+            device_option,
+        )
+        # Need to pull this out before we delete model.graph.initializer
         initialized = {init.name for init in model.graph.initializer}
+        initializer = model.graph.initializer[:]
+        # Delete the initializers so they aren't serialized
+        del model.graph.initializer[:]
+
+        cls._direct_initialize_inputs(
+            model.graph.input,
+            initialized,
+            ws,
+            device_option,
+        )
+        # Pull this out to manually add external inputs
+        external_inputs = model.graph.input[:]
+        del model.graph.input[:]
```

Comment:
hm, why do we need to delete inputs? feels wrong. But works as a hack

it might have been better to just pass a flag to the downstream functions doing conversion

Reply:
I don't like having additional flags downstream when we already have conditional checks: https://github.com/onnx/onnx-caffe2/blob/d85d53c9bcab2a9039e768ff1cc36a5b1a722c80/onnx_caffe2/backend.py#L598 https://github.com/onnx/onnx-caffe2/blob/d85d53c9bcab2a9039e768ff1cc36a5b1a722c80/onnx_caffe2/backend.py#L620